spec/solver_spec.rb

```diff
@@ -0,0 +1,24 @@
+require_relative '../solver'
+describe Solver do
+  context 'When passed an input' do
+    solver = Solver.new
+    it 'expect factorial of 5 tobe 120' do
+      expect(solver.factorial(5)).to eq 120
+    end
+    it 'expect positive number ' do
+      expect(solver.factorial(5)).to be >= 0
+    end
```

Comment:
- [ ] Based on the requirements of this project, the #factorial method should have a test that handles negative value which raises an exception as an expectation. Kindly fix :wrench: this.